core/src/typecheck/mod.rs

```diff
@@ -2420,7 +2420,39 @@ pub fn subsumption(
     checked: UnifType,
 ) -> Result<(), UnifError> {
     let inferred_inst = instantiate_foralls(state, &mut ctxt, inferred, ForallInst::UnifVar);
-    checked.unify(inferred_inst, state, &ctxt)
+    match (&inferred_inst, &checked) {
+        (
+            UnifType::Concrete {
+                typ: TypeF::Record(rrows),
+                ..
+            },
+            UnifType::Concrete {
+                typ: TypeF::Dict { type_fields, .. },
+                ..
+            },
+        ) => {
```

Comment:
Nitpick: in Rust, when you have a match case of the form (that is with only one pattern and a default case)

```rust
match foo {
  Pattern => /* .. some code */,
  _ => /* .. default case */,
}
```

It's more idiomatic to use an `if-let`. It's exactly the same result but save one level of indentation:

```rust
if let Pattern = foo then {
  /* some code */
}
else {
 /* default case */
}
```